airbyte-integrations/connectors/source-github/source_github/source.py

```diff
@@ -44,24 +44,40 @@
 )
 
 TOKEN_SEPARATOR = ","
+# To scan all the repos within orgnaization, organization name could be
+# specified by using asteriks i.e. "airbytehq/*"
+ORGANIZATION_PATTERN = re.compile("^.*/\\*$")
 
 
 class SourceGithub(AbstractSource):
     @staticmethod
-    def _generate_repositories(config: Mapping[str, Any], authenticator: MultipleTokenAuthenticator) -> List[str]:
+    def _generate_repositories(config: Mapping[str, Any], authenticator: MultipleTokenAuthenticator) -> Tuple[List[str], List[str]]:
+        """
+        Parse repositories config line and produce two lists of repositories.
+        Args:
+            config (dict): Dict representing connector's config
+            authenticator(MultipleTokenAuthenticator): authenticator object
+        Returns:
+            Tuple[List[str], List[str]]: Tuple of two lists: first representing
+            repositories directly mentioned in config and second is
+            organization repositories from orgs/{org}/repos request.
+        """
         repositories = list(filter(None, config["repository"].split(" ")))
 
         if not repositories:
             raise Exception("Field `repository` required to be provided for connect to Github API")
 
-        repositories_list = [repo for repo in repositories if not re.match("^.*/\\*$", repo)]
+        repositories_list: set = {repo for repo in repositories if not ORGANIZATION_PATTERN.match(repo)}
         organizations = [org.split("/")[0] for org in repositories if org not in repositories_list]
+        organisation_repos = set()
         if organizations:
             repos = Repositories(authenticator=authenticator, organizations=organizations)
```

Comment:
Please add a comment that states that if we receive data about an organization, then there is no need to check for each of its repositories. And add this link: https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes.

Reply:
its already there :)  https://github.com/airbytehq/airbyte/pull/8170/files#diff-895c7e2fc3c7a29570263791b1d031ceb5309127e9e42609ddd9ed425f548e47R133